### PR TITLE
Common codestyle

### PR DESCRIPTION
**Pull request by hannesa2**
With it, everyone use same codestyle

### CODE REVIEW

**Comment by eirnym** [comment]
I prefer editorconfig file with settings:

* indent_style (my personal setting is spaces)
* indent_size (my personal setting is 4)
* trim_trailing_whitespace (my personal setting is yes)
* insert_final_newline (my personal setting is yes)
* max_line_length (you set it for 150)

**Comment by hannesa2** [comment]
No problem !

Check this PR out, make your changes here
image

and commit the changed `.idea` files back to repository

**Comment by hannesa2** [comment]
But not everything can be done with it. That's why, most repositories use `ktLint` additionally

**Comment by eirnym** [comment]
`.idea` is IntelliJ-speciifc folder and can't be used by other tools

**Comment by eirnym** [comment]
> But not everything can be done with it. That's why, most repositories use `ktLint` additionally

I'd love to add a kLint and other checks globally to this repo! I've saw recently a template of gradle plugin written with Kotlin and would like to take from it as most as possible

**Comment by eirnym** [comment]
> No problem !
> 
> Check this PR out, make your changes here
> image
> 
> and commit the changed `.idea` files back to repository

I don't see that IntelliJ IDEA edits the `.editorconfig` file for a project this way

I also prefer to avoid of adding any `.idea` files into the project as we could use different versions of the editor, with different plugin versions. This would avoid incompatibilities of settings as well. Most settings we need to configure are listed in these few parameters I listed above.

**Comment by eirnym** [comment]
kLint is already have support of EditorConfig as well

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
+  <code_scheme name="Project" version="173">
+    <option name="RIGHT_MARGIN" value="150" />
```
Comment (eirnym) [inline .idea/codeStyles/Project.xml:3]: move this to `.editorconfig` file

**Review by eirnym — COMMENTED** [review]
This would at least limit number of duplicate files we need to support for each tool

**Comment by hannesa2** [comment]
Of course it's Intellij only, and in my point of view the only worth supporting IDE.
All others are out of my focus.

**Comment by eirnym** [comment]
I'd focus in kLint rather than IntelliJ IDEA.

**Comment by hannesa2** [comment]
ktlint doesn't help you inside of IDE, mostly on CI

**Comment by eirnym** [comment]
KLint is well supported in Gradle, so if you miss anything you'll get a problem

By the way, half of kLint configurations you can put into editor config file (and this is well supported). This made to support same settings across IDE even if the rest isn't supported by your editor. And I seriously doubt that klint isn't supported by IntelliJ IDEA. 

Thus you will modify 2 human-readable and human-writable files at most with no exceptions in gitignore.

**Comment by jamietanna** [comment]
Did we want to look at using [Spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle) to manage this? It works really well, auto-fixes issues, and has a lot of languages and styles supported

**Comment by eirnym** [comment]
@jamietanna looks good for me.

**Comment by jamietanna** [comment]
@hannesa2 let me know if you'd like a hand / like me to pick up applying Spotless to this project :+1:

**Comment by hannesa2** [comment]
> @hannesa2 let me know if you'd like a hand / like me to pick up applying Spotless to this project 👍

Feel free to do it.
I made already my statement with this pull request

**Comment by eirnym** [comment]
My statement is still that I prefer not to have any `.idea` folders as it's anitpattern for me. I observed too many times that different (latest) editions of IntelliJ IDEA have different project folder structure and contents (for many years we have at least 4 editions: CE, CE EAP, UE, UE EAP). 

As per other Kotlin projects, many of them have and prefer `.editorconfig` file for live editing formatting, something like [this](https://github.com/Triple-T/gradle-play-publisher/blob/master/.editorconfig). This is still most editor-agnostic way to have same formatting for Kotlin.  

Thanks to @jamietanna we have a proper formatter for java based languages (we have no other languages here)

Thus I adopt both ways of doing this: editorconfig and Spotless plugin

**Pull request closed without merging**